### PR TITLE
Add docs for Authentication.Abstractions

### PR DESCRIPTION
**Pull request by pranavkm**
Contributes to https://github.com/dotnet/aspnetcore/issues/26397

### CODE REVIEW

File: src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs [hunk 1]
```diff
@@ -122,10 +121,10 @@ public bool IsPersistent
         }
 
         /// <summary>
-        /// Set a string value in the <see cref="Items"/> collection.
+        /// Sets or remove a string value from the <see cref="Items"/> collection.
```
Comment (HaoK) [inline src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs:124]: nit: sets or removes?  Not the best at grammar but I think the plurality is supposed to be the same

File: src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs [hunk 2]
```diff
@@ -171,10 +170,10 @@ public void SetParameter<T>(string key, T value)
         }
 
         /// <summary>
-        /// Set a bool value in the <see cref="Items"/> collection.
+        /// Set or removes a <see cref="bool"/> value in the <see cref="Items"/> collection.
```
Comment (HaoK) [inline src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs:173]: set or remove?  Sets or removes? 🤷

**Review by HaoK — APPROVED** [review]

File: src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs (hunk 2 above)
Comment (pranavkm) [reply src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs:173]:
```suggestion
        /// Set or remove a <see cref="bool"/> value in the <see cref="Items"/> collection.
```

File: src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs (hunk 1 above)
Comment (pranavkm) [reply src/Http/Authentication.Abstractions/src/AuthenticationProperties.cs:124]:
```suggestion
        /// Set or remove a string value from the <see cref="Items"/> collection.
```

File: src/Http/Authentication.Abstractions/src/AuthenticationOptions.cs [hunk 3]
```diff
@@ -93,7 +96,8 @@ public void AddScheme(string name, Action<AuthenticationSchemeBuilder> configure
         public string? DefaultForbidScheme { get; set; }
 
         /// <summary>
-        /// If true, SignIn should throw if attempted with a ClaimsPrincipal.Identity.IsAuthenticated = false.
+        /// If true, SignIn should throw if attempted with a user is not authenticated.
```
Comment (Tratcher) [inline src/Http/Authentication.Abstractions/src/AuthenticationOptions.cs:99]: > /// If true, SignIn should throw if attempted with a user that is not authenticated.

File: src/Http/Authentication.Abstractions/src/AuthenticationTicket.cs [hunk 4]
```diff
@@ -55,8 +55,10 @@ public AuthenticationTicket(ClaimsPrincipal principal, string authenticationSche
 
         /// <summary>
         /// Returns a copy of the ticket.
-        /// Note: the claims principal will be cloned by calling Clone() on each of the Identities.
         /// </summary>
+        /// <remarks>
+        /// The method clones the <see cref="Principal"/> by calling <see cref="ClaimsIdentity.Clone"/> on each of the <see cref="ClaimsPrincipal.Identities"/>.
```
Comment (Tratcher) [inline src/Http/Authentication.Abstractions/src/AuthenticationTicket.cs:60]: `The <see cref="Principal"/> is cloned by calling <see cref="ClaimsIdentity.Clone"/> on each of the <see cref="ClaimsPrincipal.Identities"/>.`

**Pull request merged**